xdp-dump/test-xdpdump.sh

```diff
@@ -0,0 +1,254 @@
+
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    $XDP_LOADER load "$NS" "$TEST_PROG_DIR/test_long_func_name.o" || return 1
+
+    PID=$(start_background "$XDPDUMP -i $NS -w - 2> /dev/null | tcpdump -r - -n")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    RESULT=$(stop_background "$PID")
+
+    $XDP_LOADER unload "$NS" --all || return 1
+
+    if [[ "$RESULT" != *"$PASS_PKT"* ]]; then
+	print_result "IPv6 packet not received"
+	return 1
+    fi
+}
+
+test_capt_term()
+{
+    local PASS_REGEX="(@entry: packet size 118 bytes on if_index [0-9]+, rx queue [0-9]+)"
+    local PASS_X_REGEX="(@entry: packet size 118 bytes, captured 118 bytes on if_index [0-9]+, rx queue [0-9]+)"
+    local PASS_X_OPT="0x0020:  00 00 00 00 00 02 fc 00 de ad ca fe 00 01 00 00  ................"
+
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+
+    $XDP_LOADER load "$NS" "$TEST_PROG_DIR/test_long_func_name.o" || return 1
+
+    PID=$(start_background "$XDPDUMP -i $NS")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    RESULT=$(stop_background "$PID")
+
+    if ! [[ $RESULT =~ $PASS_REGEX ]]; then
+	print_result "IPv6 packet not received"
+	return 1
+    fi
+
+    PID=$(start_background "$XDPDUMP -i $NS -x")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    RESULT=$(stop_background "$PID")
+
+    if ! [[ $RESULT =~ $PASS_X_REGEX ]]; then
+	print_result "IPv6 packet not received[2]"
+	return 1
+    fi
+
+    # If the IP6 addresses remain the same this simple string compare can be
+    # used to verify the -x output is present.
+    if [[ "$RESULT" != *"$PASS_X_OPT"* ]]; then
+	print_result "IPv6 HEX packet not received"
+	return 1
+    fi
+
+    $XDP_LOADER unload "$NS" --all || return 1
+}
+
+test_exitentry()
+{
+    local PASS_ENTRY_REGEX="(@entry: packet size 118 bytes on if_index [0-9]+, rx queue [0-9]+)"
+    local PASS_EXIT_REGEX="(@exit\[PASS\]: packet size 118 bytes on if_index [0-9]+, rx queue [0-9]+)"
+    local PASS_EXIT_D_REGEX="(@exit\[DROP\]: packet size 118 bytes on if_index [0-9]+, rx queue [0-9]+)"
+
+    $XDP_LOADER load "$NS" "$TEST_PROG_DIR/test_long_func_name.o" || return 1
+
+    PID=$(start_background "$XDPDUMP -i $NS --rx-capture=entry")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    RESULT=$(stop_background "$PID")
+    if ! [[ $RESULT =~ $PASS_ENTRY_REGEX ]]; then
+        print_result "IPv6 entry packet not received"
+        return 1
+    fi
+
+    PID=$(start_background "$XDPDUMP -i $NS --rx-capture=exit")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" || return 1
+    RESULT=$(stop_background "$PID")
+    if ! [[ $RESULT =~ $PASS_EXIT_REGEX ]]; then
+        print_result "IPv6 exit packet not received"
+        return 1
+    fi
+
+    $XDP_LOADER unload "$NS" --all || return 1
+    $XDP_LOADER load "$NS" "$TEST_PROG_DIR/xdp_drop.o" || return 1
+
+    PID=$(start_background "$XDPDUMP -i $NS --rx-capture=exit")
+    ping6 -W 2 -c 1 "$INSIDE_IP6" # Note that this ping will fail!!
```

Comment:
Do we really need a 2-second wait for a ping we know is going to fail? Makes the test run slower than it needs to be...

Reply:
You are right, I'll update it with
"ping6 -W 0.1 -c 1 "$INSIDE_IP6" # Note that this ping will fail!!"